schemav2/resource.go

```diff
@@ -0,0 +1,136 @@
+package schemav2
+
+import (
+	"encoding/json"
+
+	"github.com/apache/arrow/go/v12/arrow/array"
+)
+
+type Resources []*Resource
+
+// Resource represents a row in it's associated table, it carries a reference to the original item, and automatically
+// generates an Id based on Table's Columns. Resource data can be accessed by the Get and Set methods
+type Resource struct {
```

Comment:
Let's update the comment on `Resource` here to talk about what it's for and how it's supposed to work with Arrow. For example, once all the fields have been set and finalized, I guess the `Build` method should be called to create an `arrow.Record` with a single row, which should then be accessed through some other method.